src/index.js

```diff
@@ -22,7 +22,7 @@ class ErrorOverlayPlugin {
         compiler.options.devServer.host
       sockOptions.sockPath =
         compiler.options.devServer.client?.webSocketURL?.pathname ||
-        compiler.options.devServer.webSocketServer?.options.path ||
+        (compiler.options.devServer.webSocketServer === "object" && compiler.options.devServer.webSocketServer.options?.path) ||
```

Comment:
because `webSocketServer` can be a string as well. `devServer: { webSocketServer: "ws" }`